ddtrace/profiling/collector/_traceback.pyx

```diff
@@ -47,9 +55,26 @@ cpdef pyframe_to_frames(frame, max_nframes):
     :param frame: The frame object to serialize.
     :param max_nframes: The maximum number of frames to return.
     :return: The serialized frames and the number of frames present in the original traceback."""
+    # DEV: There are reports that Python 3.11 returns non-frame objects when
+    # retrieving frame objects and doing stack unwinding. If we detect a
+    # non-frame object we log a warning. This until we can come to the bottom
+    # of the issue.
+    if not isinstance(frame, FrameType):
+        log.warning(
+            "Got object of type '%s' instead of a frame object for the top frame of a thread", type(frame).__name__
+        )
+        return [], 0
+
     frames = []
     nframes = 0
+
     while frame is not None:
+        if not isinstance(frame, FrameType):
+            log.warning(
+                "Got object of type '%s' instead of a frame object during stack unwinding", type(frame).__name__
+            )
+            return [], 0
```

Comment:
break ? do we want a partial if we can get it?

Reply:
I'd be more inclined to discard the event in this case, but I'll try to gather some feedback from the profiling team on this.

Reply:
actually, comment? would be good to explicitly state we want to ignore the event entirely.



oh, does it matter if we have already collected max frames?